### PR TITLE
Calculate count-weighted covariance matrix in integrate ellipsoids

### LINKED ISSUE #27789: Calculated covariance matrix in IntegrateEllipsoids does not weight by counts
For non event-mode data the covariance matrix should be weighted by the count in each bin so that the variance is more representative of the peak shape as opposed to the smaller background

### PR DESCRIPTION
**Pull request by RichardWaiteSTFC**
The covariance matrix is now weighted by the counts of a bin/event so that the variances are more representative of the peak shape and influenced less by the background.
Additionally code to find the standard deviation along the eigenvectors of the covariance matrix has been deleted - as this is simply the sqrt of the eigenvalues.

image

Example script to test (doesn't need to be WISH data but I have included the peak workspace and the diffraction MD in CountWeightedCovarianceWISH.zip in my scratch folder). SXD data will be smaller and will also work - just run FindSXPeaks to get a peak workspace.

    Load(Filename=r'\\isis.cclrc.ac.uk\inst$\NDXWISH\Instrument\data\cycle_19_1\WISH00045319.raw', OutputWorkspace='WISH000453419') 
    CropWorkspace(InputWorkspace='WISH00045319', OutputWorkspace='WISH00045319', XMin=6000, XMax=99900)
    LoadNexus(Filename='WISH_45319_peaks.nxs', OutputWorkspace='peaks')
    IntegrateEllipsoids(InputWorkspace='WISH00045319', PeaksWorkspace='peaks', SpecifySize=True, PeakSize=0.15, BackgroundInnerSize=0.15, BackgroundOuterSize=0.2, 
    OutputWorkspace='peaks_int', UseOnePercentBackgroundCorrection=False)
    # load MD to look at peak integration in slice viewer
    LoadMD(Filename='WISH_45319_MD.nxs', OutputWorkspace='WISH00045319_MD')

Note that if testing with TOPAZ event data stored as a .nxs (e.g. TOPAZ_3132_event.nxs) events may have been combined to reduce the  size (i.e. not all events will have a count of 1 and therefore this algorithm may give different results to the master branch).

Fixes #27789 

*This does not require release notes* because **it is the default behaviour that a user would expect**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/MDAlgorithms/test/IntegrateEllipsoidsWithSatellitesTest.h [hunk 1]
```diff
@@ -176,12 +222,20 @@ class IntegrateEllipsoidsWithSatellitesTest : public CxxTest::TestSuite {
...
-       * as the q-dir!
+      /* We expect an axis of the ellipsoid to be mostly
+       * along Qhkl (probably the principal axis (i.e.
+       * direction of max varaince) but not necassarily)
        */
-      TS_ASSERT_EQUALS(qDir, dirs[0]);
+      double minangle = M_PI / 2;
+      for (size_t ii = 0; ii < dirs.size(); ii++) {
+        double angle = qDir.angle(dirs[ii]);
+        if (angle > M_PI / 2) {
```
Comment (Harrietbrown) [inline Framework/MDAlgorithms/test/IntegrateEllipsoidsWithSatellitesTest.h]: why not use minangle here?

**Review by Harrietbrown — CHANGES_REQUESTED** [review]
the workspace names in the test code do not match the outputs from the loads. Are there any files in the test data or tutorial data that would substitute for `WISH_45319_peaks.nxs`?

File: Framework/MDAlgorithms/test/IntegrateEllipsoidsWithSatellitesTest.h (hunk 1 above)
Comment (RichardWaiteSTFC) [reply Framework/MDAlgorithms/test/IntegrateEllipsoidsWithSatellitesTest.h]: It should indeed be minangle, thanks!

Comment (RichardWaiteSTFC) [reply Framework/MDAlgorithms/test/IntegrateEllipsoidsWithSatellitesTest.h]: The test script has been updated - it was just a type on the run number

**Comment by Harrietbrown** [comment]
The peaks match what is seen in slice viewer, but if this was not an issue created in this release it should be in the release notes that the improvement has been made.

**Review by Harrietbrown — APPROVED** [review]
the peak indicator is better centered on the peak and better describes the shape of the peak. 

Good job.

**Pull request merged**